plugins/smartvol/transactions.go

```diff
@@ -0,0 +1,179 @@
+package smartvol
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/plugins/device/deviceutils"
+	smartvolapi "github.com/gluster/glusterd2/plugins/smartvol/api"
+
+	config "github.com/spf13/viper"
+)
+
+func txnPrepareBricks(c transaction.TxnCtx) error {
+	var req smartvolapi.Volume
+	if err := c.Get("reqBricksPrepare", &req); err != nil {
+		c.Logger().WithError(err).WithField(
+			"key", "reqBricksPrepare").Debug("Failed to get key from store")
+		return err
+	}
+
+	for _, sv := range req.Subvols {
+		for _, b := range sv.Bricks {
+			if b.PeerID != gdctx.MyUUID.String() {
+				continue
+			}
+
+			brickMountDir := path.Dir(b.Path)
+
+			// Create Mount directory
+			err := os.MkdirAll(brickMountDir, os.ModeDir|os.ModePerm)
+			if err != nil {
+				c.Logger().WithError(err).
+					WithField("path", brickMountDir).
+					Error("Failed to create brick mount directory")
+				return err
+			}
+
+			// Thin Pool Creation
+			err = deviceutils.CreateTP(b.VgName, b.TpName, b.TpSize, b.TpMetadataSize)
```

Comment:
Are we taking locks on peer before working on devices?, what if one removes the device while a device is being used by IVP module?